corehq/apps/data_dictionary/views.py

```diff
@@ -143,6 +149,133 @@ def data_dictionary_json(request, domain, case_type_name=None):
     })
 
 
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_types(request, domain):
+    fhir_resource_type_name_by_case_type = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_type_name_by_case_type = load_fhir_case_type_mapping(domain)
+
+    queryset = CaseType.objects.filter(domain=domain).annotate(properties_count=Count('property'))
+    if not request.GET.get('load_deprecated_case_types', False) == 'true':
+        queryset = queryset.filter(is_deprecated=False)
+
+    case_type_app_module_count = get_case_type_app_module_count(domain)
+    used_props_by_case_type = get_used_props_by_case_type(domain)
+    geo_case_prop = get_geo_case_property(domain)
+    case_types_data = []
+    for case_type in queryset:
+        module_count = case_type_app_module_count.get(case_type.name, 0)
+        used_props = used_props_by_case_type.get(case_type.name, [])
+        case_types_data.append({
+            "name": case_type.name,
+            "fhir_resource_type": fhir_resource_type_name_by_case_type.get(case_type),
+            "is_deprecated": case_type.is_deprecated,
+            "module_count": module_count,
+            "properties_count": case_type.properties_count,
+            "is_safe_to_delete": len(used_props) == 0,
+        })
+    return JsonResponse({
+        'case_types': case_types_data,
+        'geo_case_property': geo_case_prop,
+    })
+
+
+@login_and_domain_required
+@requires_privilege_with_fallback(privileges.DATA_DICTIONARY)
+def data_dictionary_json_case_properties(request, domain, case_type_name):
+    try:
+        skip = int(request.GET.get('skip', 0))
+        limit = int(request.GET.get('limit', 500))
+        if skip < 0 or limit < 0:
+            raise ValueError
+    except ValueError:
+        return JsonResponse({"error": _("skip and limit must be positive integers")}, status=400)
+
+    fhir_resource_prop_by_case_prop = {}
+    if toggles.FHIR_INTEGRATION.enabled(domain):
+        fhir_resource_prop_by_case_prop = load_fhir_case_properties_mapping(domain)
+
+    case_type = get_object_or_404(
+        CaseType.objects.annotate(properties_count=Count('property')),
+        domain=domain,
+        name=case_type_name
+    )
+    case_type_data = {
+        "name": case_type.name,
+        "properties_count": case_type.properties_count,
+        "groups": []
+    }
+
+    current_url = request.build_absolute_uri()
+    case_type_data["_links"] = _get_pagination_links(current_url, case_type_data["properties_count"], skip, limit)
+
+    properties_queryset = CaseProperty.objects.select_related('group').filter(case_type=case_type)
+    properties_queryset = properties_queryset.order_by('group_id', 'index', 'pk')[skip:skip + limit]
+    properties_queryset = properties_queryset.prefetch_related(
+        Prefetch('allowed_values', queryset=CasePropertyAllowedValue.objects.order_by('allowed_value'))
+    )
```

Comment:
Perhaps very subjective, but I find blocks like this to be more readable when they are inside parentheses instead of reassigned over several lines. e.g.
```suggestion
    properties_queryset = (
        CaseProperty.objects
        .select_related('group')
        .filter(case_type=case_type)
        .order_by('group_id', 'index', 'pk')[skip:skip + limit]
        .prefetch_related(Prefetch(
            'allowed_values',
            queryset=CasePropertyAllowedValue.objects.order_by('allowed_value')
        ))
    )
```

I'm okay with leaving this as it is though.

Reply:
I too feel this is more readable. Thanks Norman. Addressed in 606be7c3766acae41257a4b30e995e6cb0e98ada